osbuild/formats_common.py

```diff
@@ -0,0 +1,35 @@
+"""
+Common procedures and classes between the v1 and v2 formats
+"""
+
+from abc import abstractmethod, ABC
+from typing import Dict
+
+
+class OSBuildError(Exception, ABC):
+
+    @property
+    @abstractmethod
+    def ID(self) -> int:
+        pass
```

Comment:
This now tightly couples format specifics (error ids) into the osbuild core, since every error we want to handle has to have the `ID` field (also, capital is weird here), resp. the error code needs a `format`. This all just works for now, but as soon as we were to diverge, e.g. for the validation errors, you have to refactor everything again. Also, we also want to handle generic `Exceptions` (i.e. all random crashes) in the near future (once we have better logging). And for this it will never work.

Since we are mostly working with v2 manifests these days, I think it is totally fine to only properly format in v2 and in v2 we just map everything to a generic error. This fallback generic error we need anyway and should arguably be the first we define.